internal/proto5server/server_readresource_test.go

```diff
@@ -172,6 +174,43 @@ func TestServerReadResource(t *testing.T) {
 				NewState: testEmptyDynamicValue,
 			},
 		},
+		"request-private": {
+			server: &Server{
+				FrameworkServer: fwserver.Server{
+					Provider: &testprovider.Provider{
+						GetResourcesMethod: func(_ context.Context) (map[string]provider.ResourceType, diag.Diagnostics) {
+							return map[string]provider.ResourceType{
+								"test_resource": &testprovider.ResourceType{
+									GetSchemaMethod: func(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+										return tfsdk.Schema{}, nil
+									},
+									NewResourceMethod: func(_ context.Context, _ provider.Provider) (resource.Resource, diag.Diagnostics) {
+										return &testprovider.Resource{
+											ReadMethod: func(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {},
```

Comment:
One thing that I find very valuable during these RPC-based request field tests is validating that the provider logic (what would be inside this `ReadMethod` here) gets the expected value from the request field or raise an error diagnostic that would show up eventually as a test failure (since the diagnostic would be unexpected).

For example here:

```suggestion
											ReadMethod: func(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
												expected := "provider value"
												got, diags := req.PrivateState.GetKey("providerKey")

												resp.Diagnostics.Append(diags...)

												if string(got) != expected {
													resp.Diagnostics.AddError(
														"Unexpected req.Private Value",
														fmt.Sprintf("expected %q, got %q", expected, got),
													)
												}
											},
```